tdesktop 3.5.x and 3.6.x compilation with clang 13-14 failed

### Steps to reproduce

Build with clang 13. The previous version tdesktop worked fine.

### Expected behaviour

-

### Actual behaviour

```
[ 53%] Building CXX object Telegram/CMakeFiles/Telegram.dir/SourceFiles/api/api_hash.cpp.o
[ 53%] Building CXX object Telegram/CMakeFiles/Telegram.dir/SourceFiles/api/api_polls.cpp.o
[ 53%] Building CXX object Telegram/CMakeFiles/Telegram.dir/SourceFiles/api/api_sending.cpp.o
In file included from <built-in>:485:
In file included from /build/telegram-desktop/src/build/Telegram/CMakeFiles/Telegram.dir/cmake_pch.hxx:5:
In file included from /build/telegram-desktop/src/tdesktop-3.5.0-full/Telegram/SourceFiles/stdafx.h:102:
In file included from /usr/include/range/v3/all.hpp:17:
In file included from /usr/include/range/v3/action.hpp:32:
In file included from /usr/include/range/v3/action/split.hpp:28:
/usr/include/range/v3/range/conversion.hpp:379:29: error: alias template 'QVector' requires template arguments; argument deduction only allowed for class templates
                -> decltype(ContT(range_cpp17_iterator_t<Rng>{},
                            ^
/usr/include/meta/meta.hpp:541:5: note: in instantiation of template class 'ranges::detail::from_range<QVector>' requested here
    using invoke = typename Fn::template invoke<Args...>;
    ^
/usr/include/range/v3/range/conversion.hpp:46:13: note: in instantiation of template type alias 'invoke' requested here
            using container_t = meta::invoke<MetaFn, Rng>;
            ^
/usr/include/range/v3/range/conversion.hpp:338:46: note: in instantiation of template type alias 'container_t' requested here
                    convertible_to_cont<Rng, container_t<MetaFn, Rng>>)
                                             ^
/usr/include/range/v3/detail/prologue.hpp:35:26: note: expanded from macro 'template'
    template<__VA_ARGS__ CPP_TEMPLATE_AUX_                                                               ^
/usr/include/range/v3/range/conversion.hpp:304:13: note: in instantiation of template class 'ranges::detail::to_container::fn<ranges::detail::from_range<QVector>>' requested here
          , Fn
            ^
/usr/include/range/v3/range/conversion.hpp:425:20: note: in instantiation of template class 'ranges::detail::to_container::closure<ranges::detail::from_range<QVector>, ranges::detail::to_container::fn<ranges::detail::from_range<QVector>>>' requested here
            return detail::to_container_fn<detail::from_range<ContT>>{}(
                   ^
/build/telegram-desktop/src/tdesktop-3.5.0-full/Telegram/SourceFiles/api/api_media.cpp:86:12: note: in instantiation of function template specialization 'ranges::_to_::to<QVector, std::vector<tl::boxed<MTPinputDocument>> &>' requested here
                        ranges::to<QVector>(info.attachedStickers)),
                                ^
/usr/include/qt6/QtCore/qcontainerfwd.h:63:22: note: template is declared here
template<typename T> using QVector = QList<T>;
                     ^
In file included from <built-in>:485:
In file included from /build/telegram-desktop/src/build/Telegram/CMakeFiles/Telegram.dir/cmake_pch.hxx:5:
In file included from /build/telegram-desktop/src/tdesktop-3.5.0-full/Telegram/SourceFiles/stdafx.h:102:
In file included from /usr/include/range/v3/all.hpp:17:
In file included from /usr/include/range/v3/action.hpp:32:
In file included from /usr/include/range/v3/action/split.hpp:28:
/usr/include/range/v3/range/conversion.hpp:425:20: error: type 'detail::to_container_fn<detail::from_range<QVector>>' (aka 'closure<ranges::detail::from_range<QVector>, ranges::detail::to_container::fn<ranges::detail::from_range<QVector>>>') does not provide a call operator
            return detail::to_container_fn<detail::from_range<ContT>>{}(
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/telegram-desktop/src/tdesktop-3.5.0-full/Telegram/SourceFiles/api/api_media.cpp:86:12: note: in instantiation of function template specialization 'ranges::_to_::to<QVector, std::vector<tl::boxed<MTPinputDocument>> &>' requested here
                        ranges::to<QVector>(info.attachedStickers)),
                                ^
2 errors generated.
make[2]: *** [Telegram/CMakeFiles/Telegram.dir/build.make:493: Telegram/CMakeFiles/Telegram.dir/SourceFiles/api/api_media.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:1550: Telegram/CMakeFiles/Telegram.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
==> ERROR: Build failed, check /build
```

### Operating system

Linux

### Version of Telegram Desktop

3.5.0

### Installation source

Other (unofficial) source

### Logs

_No response_

--- a/Telegram/SourceFiles/api/api_media.cpp
+++ b/Telegram/SourceFiles/api/api_media.cpp
@@ -83,7 +83,7 @@ MTPInputMedia PrepareUploadedPhoto(RemoteFileInfo info) {
 		MTP_flags(flags),
 		info.file,
 		MTP_vector<MTPInputDocument>(
-			ranges::to<QVector>(info.attachedStickers)),
+			ranges::to<QVector<MTPInputDocument>>(info.attachedStickers)),
 		MTP_int(0));
 }
 
@@ -107,7 +107,7 @@ MTPInputMedia PrepareUploadedDocument(
 		MTP_string(document->mimeString()),
 		ComposeSendingDocumentAttributes(document),
 		MTP_vector<MTPInputDocument>(
-			ranges::to<QVector>(info.attachedStickers)),
+			ranges::to<QVector<MTPInputDocument>>(info.attachedStickers)),
 		MTP_int(0));
 }
 

--- a/Telegram/SourceFiles/boxes/peers/edit_peer_reactions.cpp
+++ b/Telegram/SourceFiles/boxes/peers/edit_peer_reactions.cpp
@@ -138,7 +138,7 @@ void SaveAllowedReactions(
 		const std::vector<QString> &allowed) {
 	auto ids = allowed | ranges::views::transform([=](QString value) {
 		return MTP_string(value);
-	}) | ranges::to<QVector>;
+	}) | ranges::to<QVector<MTPstring>>;
 
 	peer->session().api().request(MTPmessages_SetChatAvailableReactions(
 		peer->input,

